src/taco-dependency-installer/installers/androidPackagesInstaller.ts

```diff
@@ -0,0 +1,296 @@
+﻿/**
+﻿ *******************************************************
+﻿ *                                                     *
+﻿ *   Copyright (C) Microsoft. All rights reserved.     *
+﻿ *                                                     *
+﻿ *******************************************************
+﻿ */
+
+/// <reference path="../../typings/dependencyInstallerInterfaces.d.ts" />
+/// <reference path="../../typings/Q.d.ts" />
+
+"use strict";
+
+import childProcess = require ("child_process");
+import fs = require ("fs");
+import os = require ("os");
+import path = require ("path");
+import Q = require ("q");
+
+import InstallerBase = require ("./installerBase");
+import installerProtocol = require ("../elevatedInstallerProtocol");
+import resources = require ("../resources/resourceManager");
+import tacoUtils = require ("taco-utils");
+
+import ILogger = installerProtocol.ILogger;
+import utilHelper = tacoUtils.UtilHelper;
+
+class AndroidPackagesInstaller extends InstallerBase {
+    private static ANDROID_HOME_NAME: string = "ANDROID_HOME";
+    private static PLATFORM_TOOLS_PKG: string = "platform-tools";
+    private static BUILD_TOOLS_PKG_PREFIX: string = "build-tools-";
+
+    private androidCommand: string;
+    private adbCommand: string;
+    private availablePackages: string[];
+
+    constructor(installerInfo: DependencyInstallerInterfaces.IInstallerData, softwareVersion: string, installTo: string, logger: ILogger, steps: DependencyInstallerInterfaces.IStepsDeclaration) {
+        super(installerInfo, softwareVersion, installTo, logger, steps, "androidPackages");
+    }
+
+    protected installWin32(): Q.Promise<any> {
+        return this.installDefault();
+    }
+
+    protected installDarwin(): Q.Promise<any> {
```

Comment:
The install method is another good candidate for going into a new platform interface. Here, you call installDefault() for both, which is a clear sign that you are abstracting the implementation at a lower level, not at the higher level of your class architecture.